### PR TITLE
Add entry point for running titus-executor from the VK

### PR DESCRIPTION
**Pull request by gabrielhartmann**
### Description of the Change

This PR is a child of the `vk3` branch which has been modified over time to support use by the Titus Virtual Kubelet implementation.

### CODE REVIEW

File: cmd/virtual-kubelet-backend-titus-executor/main.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	"github.com/Netflix/titus-executor/executor/runtime/docker"
+	"github.com/Netflix/titus-executor/vk/backend"
+
+	"github.com/Netflix/titus-executor/uploader"
+	"github.com/sirupsen/logrus"
+	"github.com/virtual-kubelet/virtual-kubelet/log"
+	logruslogger "github.com/virtual-kubelet/virtual-kubelet/log/logrus"
+	"gopkg.in/urfave/cli.v1"
+	"time"
+
```
Comment (sargun) [inline cmd/virtual-kubelet-backend-titus-executor/main.go]: something seems weird with your gofmt?

File: cmd/virtual-kubelet-backend-titus-executor/main.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		},
+		cli.StringFlag{
+			Name: "status",
+			Destination:&statusPipe,
+			Usage: "The location of the status pipe",
+		},
+	}
+
+	app := cli.NewApp()
+	app.Name = "titus-executor"
```
Comment (sargun) [inline cmd/virtual-kubelet-backend-titus-executor/main.go]: should the app name be something like `titus-executor-virtual-kubelet-backend`?

File: cmd/virtual-kubelet-backend-titus-executor/main.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+func mainWithError(podFileName string, statusPipe string, dockerCfg *docker.Config, cfg *config.Config) error {
+	logrus.SetLevel(logrus.DebugLevel)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	logger := logrus.StandardLogger()
+	logger.SetLevel(logrus.DebugLevel)
+	log.L = logruslogger.FromLogrus(logrus.NewEntry(logger))
+	ctx = log.WithLogger(ctx, log.L)
+
+	log.G(ctx).Debugf("Args: %v, %v", podFileName, statusPipe)
```
Comment (sargun) [inline cmd/virtual-kubelet-backend-titus-executor/main.go]: Please use withField

File: cmd/virtual-kubelet-backend-titus-executor/main.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	log.G(ctx).Debugf("os.Args[0]: %v", os.Args[0])
```
Comment (sargun) [inline cmd/virtual-kubelet-backend-titus-executor/main.go]: Please use %s, or withfield.

Comment (sargun) [reply cmd/virtual-kubelet-backend-titus-executor/main.go]: or %q

File: cmd/virtual-kubelet-backend-titus-executor/main.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+	if err != nil {
+		panic(err)
+	}
+	log.G(ctx).WithField("pod", pod.Name).Debugf("Got pipe %v", statusPipe)
+
+	logUploaders := uploader.NewUploadersFromUploaderArray([]uploader.Uploader{})
+
+	dockerRunner, err := runner.New(ctx, metrics.Discard, logUploaders, *cfg, *dockerCfg)
+	if err != nil {
+		return fmt.Errorf("cannot create Titus executor: %v", err)
```
Comment (sargun) [inline cmd/virtual-kubelet-backend-titus-executor/main.go]: errors.wrapf will achieve this as well, but in a somewhat more standardized way

File: vk/backend/backend.go [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+		return v1.PodFailed
+	default:
+		panic(state)
+	}
+}
+
+func RunWithBackend(ctx context.Context, runner *runner.Runner, statuses *os.File, pod *v1.Pod) error {
+	containerInfoStr, ok := pod.GetAnnotations()["containerInfo"]
+	if !ok {
+		return errors.New("Cannot find container info")
```
Comment (sargun) [inline vk/backend/backend.go]: These should be defined up above, IMHO.

Comment (sargun) [reply vk/backend/backend.go]: as like 
```
var (
errfoo = errors.New(...)
)
```

File: vk/backend/backend.go [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+	if err != nil {
+		return errors.Wrap(err, "Could not start task")
+	}
+
+	encoder := json.NewEncoder(statuses)
+	var lock sync.Mutex
+
+	go func() {
+		ch := make(chan os.Signal, 1)
+		signal.Notify(ch, os.Interrupt, unix.SIGUSR1, unix.SIGHUP, unix.SIGTERM)
```
Comment (sargun) [inline vk/backend/backend.go]: what is sighup supposed to do?

Comment (gabrielhartmann) [reply vk/backend/backend.go]: Forgot to remove this.  I had code in here before that would resend a TaskStatus (pod) on SIGHUP.

File: cmd/virtual-kubelet-backend-titus-executor/main.go (hunk 5 above)
Comment (gabrielhartmann) [reply cmd/virtual-kubelet-backend-titus-executor/main.go]: errors.Wrapf is not welcome according to `go test`.  I have no idea why but its inclusion in ```backend/helpers.go``` was crashing `go test`, so I'm reticent to include it again.  I know it's used elsewhere in the project...  This has eaten enough hours in my opinion.

Comment (gabrielhartmann) [reply cmd/virtual-kubelet-backend-titus-executor/main.go]: I'll try it anyway...

**Pull request merged**